syspagegen/syspagegen.c

```diff
@@ -0,0 +1,822 @@
+					(strcmp(name, CAST_SYSPTR(char *, map->name)) == 0))
+				return -1;
+
+		} while ((ptr = map->next) != sysgen_common.syspage->maps);
+	}
+
+	return 0;
+}
+
+
+static int sysgen_mapNameResolve(const char *name, uint8_t *id)
+{
+	sysptr_t ptr;
+	const syspage_map_t *map;
+
+	if (sysgen_common.syspage->maps != 0) {
+		ptr = sysgen_common.syspage->maps;
+		do {
+			map = CAST_SYSPTR(syspage_map_t *, ptr);
+
+			if (strcmp(name, CAST_SYSPTR(char *, map->name)) == 0) {
+				*id = map->id;
+				return 0;
+			}
+
+		} while ((ptr = map->next) != sysgen_common.syspage->maps);
+	}
+
+	return -1;
+}
+
+
+static int sysgen_strAttr2ui(const char *str, unsigned int *attr)
+{
+	int i;
+
+	*attr = 0;
+	for (i = 0; str[i]; ++i) {
+		switch (str[i]) {
+			case 'r':
+				*attr |= mAttrRead;
+				break;
+			case 'w':
+				*attr |= mAttrWrite;
+				break;
+			case 'x':
+				*attr |= mAttrExec;
+				break;
+			case 's':
+				*attr |= mAttrShareable;
+				break;
+			case 'c':
+				*attr |= mAttrCacheable;
+				break;
+			case 'b':
+				*attr |= mAttrBufferable;
+				break;
+			default:
+				fprintf(stderr, "\nsysgen: Wrong attribute - '%c'", str[i]);
+				return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+static int sysgen_mapAdd(const char *mapName, addr_t start, addr_t end, unsigned int attr)
+{
+	size_t len;
+	char *name;
+	sysptr_t ptr;
+	syspage_map_t *map, *headMap;
+
+	ptr = sysgen_buffAlloc(sizeof(syspage_map_t));
+	if (ptr == 0)
+		return -1;
+
+	map = CAST_SYSPTR(syspage_map_t *, ptr);
+
+	len = strlen(mapName);
+	map->name = sysgen_buffAlloc(len + 1);
+	if (map->name == 0)
+		return -1;
+
+	name = CAST_SYSPTR(char *, map->name);
+	memcpy(name, mapName, len);
+	name[len] = '\0';
+
+	map->entries = 0;
+	map->start = start;
+	map->end = end;
+	map->attr = attr;
+
+	if (sysgen_common.syspage->maps == 0) {
+		map->next = ptr;
+		map->prev = ptr;
+		sysgen_common.syspage->maps = ptr;
+	}
+	else {
+		headMap = CAST_SYSPTR(syspage_map_t *, sysgen_common.syspage->maps);
+
+		map->prev = headMap->prev;
+		CAST_SYSPTR(syspage_map_t *, headMap->prev)->next = ptr;
+		map->next = sysgen_common.syspage->maps;
+		headMap->prev = ptr;
+	}
+
+	map->id = (ptr == sysgen_common.syspage->maps) ? 0 : CAST_SYSPTR(syspage_map_t *, map->prev)->id + 1;
+
+	return 0;
+}
+
+
+int sysgen_cmdMap(int argc, char *argv[])
+{
+	int res;
+	char *endptr;
+	addr_t start, end;
+	unsigned int attr;
+
+	if (argc != 5) {
+		fprintf(stderr, "\n%s: Wrong argument count", argv[0]);
+		return -1;
+	}
+
+	start = strtoul(argv[2], &endptr, 0);
+	if (*endptr) {
+		fprintf(stderr, "\n%s: Wrong arguments", argv[0]);
+		return -1;
+	}
+
+	end = strtoul(argv[3], &endptr, 0);
+	if (*endptr) {
+		fprintf(stderr, "\n%s: Wrong arguments", argv[0]);
+		return -1;
+	}
+	if ((res = sysgen_strAttr2ui(argv[4], &attr)) < 0)
+		return res;
+
+	/* Check whether map's name exists or map overlaps with other maps */
+	res = sysgen_mapOverlapping(argv[1], start, end);
+	if (res < 0)
+		return res;
```

Comment:
Error checking style mix

```suggestion
	res = sysgen_strAttr2ui(argv[4], &attr);
	if (res < 0)
		return res;

	/* Check whether map's name exists or map overlaps with other maps */
	res = sysgen_mapOverlapping(argv[1], start, end);
	if (res < 0)
		return res;
```

Reply:
Done